internal/callbacker/send_manager.go

```diff
@@ -67,46 +67,85 @@ const (
 	ActiveMode
 	QuarantineMode
 	StoppingMode
+
+	entriesBufferSize = 10000
 )
 
-func runNewSendManager(url string, c SenderI, s store.CallbackerStore, l *slog.Logger, q *quarantinePolicy,
-	sendDelay, singleSendSleep, batchSendInterval time.Duration) *sendManager {
-	const defaultBatchSendInterval = time.Duration(5 * time.Second)
-	if batchSendInterval == 0 {
-		batchSendInterval = defaultBatchSendInterval
+func WithBufferSize(size int) func(*sendManager) {
+	return func(m *sendManager) {
+		m.entries = make(chan *CallbackEntry, size)
+		m.batchEntries = make(chan *CallbackEntry, size)
+	}
+}
+
+func runNewSendManager(url string, sender SenderI, store store.CallbackerStore, logger *slog.Logger, quarantinePolicy *quarantinePolicy, sendingConfig *SendConfig, opts ...func(*sendManager)) *sendManager {
+	const defaultBatchSendInterval = 5 * time.Second
+
+	batchSendInterval := defaultBatchSendInterval
+	if sendingConfig.BatchSendInterval != 0 {
+		batchSendInterval = sendingConfig.BatchSendInterval
 	}
 
 	m := &sendManager{
-		url:        url,
-		sender:     c,
-		s:          s,
-		l:          l,
-		quarantine: q,
-
-		sendDelay:         sendDelay,
-		singleSendSleep:   singleSendSleep,
+		url:              url,
+		sender:           sender,
+		store:            store,
+		logger:           logger,
+		quarantinePolicy: quarantinePolicy,
+
+		sendDelay:         sendingConfig.Delay,
+		singleSendSleep:   sendingConfig.PauseAfterSingleModeSuccessfulSend,
 		batchSendInterval: batchSendInterval,
 
-		entries:      make(chan *CallbackEntry),
-		batchEntries: make(chan *CallbackEntry),
+		entries:      make(chan *CallbackEntry, entriesBufferSize),
+		batchEntries: make(chan *CallbackEntry, entriesBufferSize),
 		stop:         make(chan struct{}),
 	}
 
+	for _, opt := range opts {
+		opt(m)
+	}
+
 	m.run()
 	return m
 }
 
 func (m *sendManager) Add(entry *CallbackEntry, batch bool) {
-	m.entriesWg.Add(1) // count the callbacks accepted for processing
-	go func() {
```

Comment:
Why would this lead to abandoning callbacks?

Reply:
Maybe the consumption of single callbacks here is not fast enough
https://github.com/bitcoin-sv/arc/blob/0629dab3a7fb6305a8ecfd8d9ff7129e091e6ad8/internal/callbacker/send_manager.go#L155

and when callbacker restarts all open go routines could be lost. Not 100% sure it is the problem. But the unbuffered channel is the only thing which looks like it could be the issue.